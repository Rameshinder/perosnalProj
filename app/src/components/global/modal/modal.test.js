import React from 'react'
import { shallow, mount } from 'enzyme'
import Modal from './index'
jest.mock('../../icon', () => 'Icon')
jest.mock('@anz/button', () => 'Button')
describe('Modal', () => {
  const app = shallow(<Modal />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })
  it('Modal should render without crashing', () => {
    shallow(<Modal />)
  })
  it('Modal should call closeModal on click of close', () => {
    shallow(<Modal />)
  })
  it('should clear the search field when cross icon is clicked', () => {
    const spy = jest.spyOn(Modal.prototype, 'closeModal')
    const wrapper = mount(<Modal title='test modal' show onClose={jest.fn()} />)
    const searchIcon = wrapper.find('a')
    searchIcon.simulate('click')
    expect(spy).toHaveBeenCalled()
  })
  it('Footer should render without crashing', () => {
    shallow(<Modal />)
  })
  it('renders sendFooterText and cancelFooterText props', () => {
    const wrapper = mount(<Modal
      show={false}
      cancelFooterText=' '
      sendFooterText=' '
      title='Feedback'
      onClose={jest.fn()}
      disabled
      handleClick={jest.fn()}
      height=' '
      width=' ' />)
    wrapper.setProps({sendFooterText: null, cancelFooterText: null, height: '1100', width: '738', addAnimation: 'slidedown'})
    expect(wrapper.prop('sendFooterText')).toBe(null)
    expect(wrapper.prop('cancelFooterText')).toBe(null)
    expect(wrapper.prop('height')).toBe('1100')
    expect(wrapper.prop('width')).toBe('738')
    expect(wrapper.prop('addAnimation')).toBe('slidedown')
  })
  test('handleClick', () => {
    const handleClickProp = jest.fn()
    const wrapper = mount(<Modal
      show
      cancelFooterText='Cancel'
      sendFooterText='Send'
      title='Feedback'
      onClose={jest.fn()}
      disabled
      handleClick={handleClickProp}
      height='100'
      width='400' />
    )
    wrapper.instance().handleClick()
    expect(handleClickProp).toHaveBeenCalled()
  })
})
