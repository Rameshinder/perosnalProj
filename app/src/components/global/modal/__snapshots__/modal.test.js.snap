// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    disableCancel={true}
    disableSend={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        addAnimation={undefined}
        height={undefined}
        role="dialog"
        tabIndex={0}
        title="feedbackModal"
        width={undefined}
      >
        <styled.div>
          <Heading
            aria-label={undefined}
            level="2"
            tabIndex={0}
          />
          <styled.a
            href="javascript:void(0)"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={0}
            title="Close"
          >
            <Icon
              color="#8e8e8e"
              iconName="Cross"
            />
          </styled.a>
        </styled.div>
        <styled.hr />
        <styled.div />
        <styled.div />
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addAnimation": undefined,
        "children": Array [
          <styled.div>
            <Heading
              aria-label={undefined}
              level="2"
              tabIndex={0}
            />
            <styled.a
              href="javascript:void(0)"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={0}
              title="Close"
            >
              <Icon
                color="#8e8e8e"
                iconName="Cross"
              />
            </styled.a>
          </styled.div>,
          <styled.hr />,
          <styled.div />,
          <styled.div />,
        ],
        "height": undefined,
        "role": "dialog",
        "tabIndex": 0,
        "title": "feedbackModal",
        "width": undefined,
      },
      "ref": "modalCmp",
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Heading
                aria-label={undefined}
                level="2"
                tabIndex={0}
              />,
              <styled.a
                href="javascript:void(0)"
                onClick={[Function]}
                onKeyPress={[Function]}
                tabIndex={0}
                title="Close"
              >
                <Icon
                  color="#8e8e8e"
                  iconName="Cross"
                />
              </styled.a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "aria-label": undefined,
                "children": undefined,
                "level": "2",
                "tabIndex": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Icon
                  color="#8e8e8e"
                  iconName="Cross"
                />,
                "href": "javascript:void(0)",
                "onClick": [Function],
                "onKeyPress": [Function],
                "tabIndex": 0,
                "title": "Close",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "color": "#8e8e8e",
                  "iconName": "Cross",
                },
                "ref": null,
                "rendered": null,
                "type": "Icon",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              null,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          addAnimation={undefined}
          height={undefined}
          role="dialog"
          tabIndex={0}
          title="feedbackModal"
          width={undefined}
        >
          <styled.div>
            <Heading
              aria-label={undefined}
              level="2"
              tabIndex={0}
            />
            <styled.a
              href="javascript:void(0)"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={0}
              title="Close"
            >
              <Icon
                color="#8e8e8e"
                iconName="Cross"
              />
            </styled.a>
          </styled.div>
          <styled.hr />
          <styled.div />
          <styled.div />
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addAnimation": undefined,
          "children": Array [
            <styled.div>
              <Heading
                aria-label={undefined}
                level="2"
                tabIndex={0}
              />
              <styled.a
                href="javascript:void(0)"
                onClick={[Function]}
                onKeyPress={[Function]}
                tabIndex={0}
                title="Close"
              >
                <Icon
                  color="#8e8e8e"
                  iconName="Cross"
                />
              </styled.a>
            </styled.div>,
            <styled.hr />,
            <styled.div />,
            <styled.div />,
          ],
          "height": undefined,
          "role": "dialog",
          "tabIndex": 0,
          "title": "feedbackModal",
          "width": undefined,
        },
        "ref": "modalCmp",
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Heading
                  aria-label={undefined}
                  level="2"
                  tabIndex={0}
                />,
                <styled.a
                  href="javascript:void(0)"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  tabIndex={0}
                  title="Close"
                >
                  <Icon
                    color="#8e8e8e"
                    iconName="Cross"
                  />
                </styled.a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "aria-label": undefined,
                  "children": undefined,
                  "level": "2",
                  "tabIndex": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Icon
                    color="#8e8e8e"
                    iconName="Cross"
                  />,
                  "href": "javascript:void(0)",
                  "onClick": [Function],
                  "onKeyPress": [Function],
                  "tabIndex": 0,
                  "title": "Close",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "color": "#8e8e8e",
                    "iconName": "Cross",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "Icon",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                null,
                null,
              ],
            },
            "ref": null,
            "rendered": Array [
              null,
              null,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
